compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -923,7 +923,51 @@ trait Implicits:
           // example where searching for a nested type causes an infinite loop.
           None
 
-    MissingImplicitArgument(arg, pt, where, paramSymWithMethodCallTree, ignoredInstanceNormalImport)
+    def allImplicits(currImplicits: ContextualImplicits): List[ImplicitRef] =
+      if currImplicits.outerImplicits == null then currImplicits.refs
+      else currImplicits.refs ::: allImplicits(currImplicits.outerImplicits)
+
+    /** Ensure an implicit is not a Scala 2-style implicit conversion, based on its type */
+    def isScala2Conv(typ: Type): Boolean = typ match {
+      case PolyType(_, resType) => isScala2Conv(resType)
+      case mt: MethodType => !mt.isImplicitMethod && !mt.isContextualMethod
+      case _ => false
+    }
+
+    def hasErrors(tree: Tree): Boolean =
+      if tree.tpe.isInstanceOf[ErrorType] then true
+      else
+        tree match {
+          case Apply(_, List(arg)) => arg.tpe.isInstanceOf[ErrorType]
+          case _ => false
+        }
+
+    def ignoredConvertibleImplicits = arg.tpe match
+      case fail: SearchFailureType =>
+        if (fail.expectedType eq pt) || isFullyDefined(fail.expectedType, ForceDegree.none) then
+          // Get every implicit in scope and try to convert each
+          allImplicits(ctx.implicits)
+            .distinctBy(_.underlyingRef.denot)
+            .view
+            .map(_.underlyingRef)
+            .filter { imp =>
```

Comment:
Actually, what was the problem with using `ctx.implicits.eligible(ViewProto(...))` to find out if an implicit value can be converted to the type that we're searching for?

Reply:
I was thinking that in case the implicit value was something like `given [T <: Bar]: Foo[T]`, it would've needed `wildApprox` to fill in the type argument to `Foo` and `Foo[?]` would've been too permissive, but after your comment, I just realized `wildApprox` takes bounds into account. It'd be clearer than the current approach and would give us the conversions that would apply to the implicits. I'll try that later.

Reply:
So I tried the following to check if there are applicable conversions:
```scala
ctx.implicits.eligible(
  ViewProto(
    wildApprox(imp.underlying.widen.finalResultType),
    fail.expectedType
  )
).nonEmpty
```
This works fine when the conversion doesn't have any type parameters (`summon[String]` and `implicitly[String]` work fine in the test), but when it does have type parameters, all the implicits in scope are listed (`summon[Option[Int]]` and `implicitly[Option[Int]]` don't work). Gonna have to do some more investigating there.

Reply:
I tried to dig into that and it seems that we were trying to reinvent the wheel. There's already the method `viewExists` available in this context, which seems to does what `canBeConverted` was supposed to do` and it even has the same signature

Reply:
Nice find!